controllers/module_reconciler.go

```diff
@@ -440,9 +428,55 @@ func (mrh *moduleReconcilerHelper) setKMMOMetrics(ctx context.Context) {
 	err := mrh.client.List(ctx, &mods)
 	if err != nil {
 		logger.V(1).Info("failed to list KMMomodules for metrics", "error", err)
-	}
+		return
+	}
+
+	numModules := len(mods.Items)
+	numModulesWithBuild := 0
+	numModulesWithSign := 0
+	numModulesWithDevicePlugin := 0
+	for _, mod := range mods.Items {
+		signFound := false
+		buildFound := false
+		if mod.Spec.DevicePlugin != nil {
+			numModulesWithDevicePlugin += 1
+		}
+		if mod.Spec.ModuleLoader.Container.Build != nil {
+			buildFound = true
+		}
+		if mod.Spec.ModuleLoader.Container.Sign != nil {
+			signFound = true
+		}
+		if !buildFound || !signFound {
+			for _, mapping := range mod.Spec.ModuleLoader.Container.KernelMappings {
+				if mapping.Sign != nil {
+					signFound = true
+				}
+				if mapping.Build != nil {
+					buildFound = true
+				}
+			}
+		}
```

Comment:
How about encapsulating in a function/method?
We can have a method for build that leverages `ShouldBeBuilt` and another one for sign that uses `ShouldBeSigned`. 
WDYT?